flowredux/src/commonMain/kotlin/com/freeletics/flowredux/dsl/BaseBuilderBlock.kt

```diff
@@ -151,6 +151,31 @@ public abstract class BaseBuilderBlock<InputState : S, S : Any, A : Any> interna
         }
     }
 
+    /**
+     * Triggers every time the state machine enters this state. The passed [Flow] created by
+     * [flowBuilder] will be collected and any emission will be passed to [handler].
+     *
+     * The collection as well as any ongoing [handler] is cancelled when leaving this state.
+     *
+     * [executionPolicy] is used to determine the behavior when a new emission from `Flow` arrives
+     * before the previous [handler] invocation completed. By default [ExecutionPolicy.ORDERED]
+     * is applied.
+     */
+    public fun <T> collectWhileInStateBasedOnState(
+        flowBuilder: (InputState) -> Flow<T>,
+        executionPolicy: ExecutionPolicy = ExecutionPolicy.ORDERED,
+        handler: suspend (item: T, state: State<InputState>) -> ChangedState<S>,
+    ) {
+        sideEffectBuilders += SideEffectBuilder(isInState) { initialState ->
+            CollectWhile(
+                isInState = sideEffectIsInState(),
```

Comment:
Need to update this when the identity pr is merged

```suggestion
                isInState = sideEffectIsInState(initialState),
```